pkg/roachpb/api.proto

```diff
@@ -1633,6 +1633,16 @@ message AddSSTableRequest {
   // Added in 22.1, so check the MVCCAddSSTable version gate before using.
   bool write_at_request_timestamp = 6;
 
+  // SSTTimestamp is a promise from the client that all MVCC timestamps in the
```

Comment:
I wonder if we should fold this into write_at_request_timestamp, so the API becomes "update this placeholder timestamp to the request timestamp" rather than "update to the request timestamp" and "this is the placeholder timestamp" ?

Reply:
I think it's fine to keep these independent -- I think it should be possible to use `WriteAtRequestTimestamp` and just pass in any random SST with arbitrary timestamps, and still have it work as expected. It also makes the API a bit clearer, I think.

Reply:
oh, now I remember why I was thinking this: The fast pebble rewriter currently demands you tell it what suffix you want replaced (`from, to []byte`) so to move it to the batch ts, we want to know what ts it has to start, thus thinking that "SSTTimestamp` and `WriteAtBatchTS` become just `move ts from x to batch ts`